homeassistant/components/zha/__init__.py

```diff
@@ -192,11 +194,14 @@ def handle_message(sender, is_reply, profile, cluster,
 
     api.async_load_api(hass, application_controller, zha_gateway)
 
-    def zha_shutdown(event):
-        """Close radio."""
+    async def async_zha_shutdown(event):
+        """Handle shutdown tasks."""
+        await hass.data[DATA_ZHA][
+            DATA_ZHA_GATEWAY].async_update_device_storage()
         hass.data[DATA_ZHA][DATA_ZHA_RADIO].close()
```

Comment:
Is this async? Because it used to be a sync function calling this. 
If it's async friendly, prefix method with `async_`

Reply:
The close method? it's in bellows so I can't prefix it. 

https://github.com/zigpy/bellows/blob/e1cf24e7b4081c546ddf4e8c67165f8fec1c6898/bellows/ezsp.py#L41

https://github.com/zigpy/bellows/blob/a5b0fd074862c14f5a656ca306c5e6193d575558/bellows/uart.py#L154

It should be safe to call from async. I have been running this for a week with no issues. I can kick it to the executor to be safe if you prefer?

Reply:
It closes the serial port. Does not cause IO blocking.